azureappconfiguration/internal/jsonc/jsonc.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Package jsonc provides support for parsing JSON with comments (JSONC).
+// This package extends standard JSON support to handle line comments (//) and block comments (/* */).
+package jsonc
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// StripComments removes comments from JSONC while preserving string literals.
+func StripComments(data []byte) []byte {
+	var result []byte
+	var inString bool
+	var escaped bool
+
+	for i := 0; i < len(data); i++ {
+		char := data[i]
+
+		if escaped {
+			// Previous character was a backslash, so this character is escaped
+			result = append(result, char)
+			escaped = false
+			continue
+		}
+
+		if inString {
+			result = append(result, char)
+			if char == '\\' {
+				escaped = true
+			} else if char == '"' {
+				inString = false
+			}
+			continue
+		}
+
+		// Not in a string
+		if char == '"' {
+			inString = true
+			result = append(result, char)
+			continue
+		}
+		// Check for line comment
+		if char == '/' && i+1 < len(data) && data[i+1] == '/' {
+			// Skip until end of line
+			i++ // skip the second '/'
+			for i < len(data) && data[i] != '\n' && data[i] != '\r' {
```

Comment:
This comment was made in my pr, and I'm unsure. Do we need to check for `\r` windows escapes are `\r\n` which using this logic still works without the  `\r` check.

Reply:
By "Old Macs" it means Mac OS 9, which was end of life some 25 years ago. I think it's reasonable to assume we don't support this, you would have to go out of your way to try and break this.